java/src/com/github/bazel_contrib/contrib_rules_jvm/javaparser/generators/ClasspathParser.java

```diff
@@ -284,23 +295,39 @@ public Void visitVariable(VariableTree node, Void unused) {
       return super.visitVariable(node, unused);
     }
 
-    private void checkFullyQualifiedType(Tree identifier) {
+    @Nullable
+    private Set<String> checkFullyQualifiedType(Tree identifier) {
+      Set<String> types = new TreeSet<>();
       if (identifier.getKind() == Tree.Kind.IDENTIFIER
           || identifier.getKind() == Tree.Kind.MEMBER_SELECT) {
         String typeName = identifier.toString();
-        if (typeName.contains(".")) {
+        List<String> components = Splitter.on(".").splitToList(typeName);
+        if (currentFileImports.containsKey(components.get(0))) {
+          String importedType = currentFileImports.get(components.get(0));
+          usedTypes.add(importedType);
+          types.add(importedType);
+        } else if (components.size() > 1) {
           usedTypes.add(typeName);
+          types.add(typeName);
+        } else {
+          String expandedImport = currentFileImports.get(typeName);
+          if (expandedImport != null) {
+            usedTypes.add(expandedImport);
+            types.add(expandedImport);
+          }
```

Comment:
I don't know when this branch will trigger. If I'm reading this correctly, we have:
- `if (currentFileImports.containsKey(components.get(0))) {` -> Will trigger for both `IDENTIFIER` and `MEMBER_SELECT` where we've already imported the left hand side.
- ` else if (components.size() > 1) {` -> Unconditionally import all other `MEMBER_SELECT`s, because they're by definition not in the currently used imports.

It feels like the only types left for this `else` clause are `IDENTIFIER`s that are _not_ already in `currentFileImports`, so this `get` will always return `null`.

Reply:
You're right, I don't think it can happen - removed